src/include/network/lan/address.rb

```diff
@@ -576,22 +581,14 @@ def StoreVLANSlave(_key, _event)
       nil
     end
 
+    # Given a device name returns the position in the bond slaves table
+    # or -1 if not present
+    #
+    # @param [String] slave name
+    # @return [Integer] index of the given slave in msbox_items table
     def getISlaveIndex(slave)
-      items = Convert.convert(
-        UI.QueryWidget(:msbox_items, :Items),
-        from: "any",
-        to:   "list <term>"
-      )
-      index = -1
-      pos = 0
-      Builtins.foreach(items) do |it|
-        if Ops.get_string(it, [0, 0], "") == slave
-          index = pos
-          raise Break
-        end
-        pos = Ops.add(pos, 1)
-      end
-      index
+      items = UI.QueryWidget(:msbox_items, :Items)
+      items.index { |it| Ops.get_string(it, [0, 0], "") == slave } || -1
```

Comment:
You are right, and i changed it initially but i don't like at all 

``` it.fetch(0, []).fetch(0, "") ```

 is not so legible or 

```(it[0] || [])[0] || "" ```

It will be nice with new dig method.